### PR TITLE
[UTOPIA-550,552]add url route to view pia intake detail

### PR DESCRIPTION
**Pull request by binggaofw**
# Description

This PR includes the following proposed change(s):

- add new front route to view pia intake details
Screen Shot 2022-12-16 at 10 33 38 AM

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring / Documentation
- [ ] Version change

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

## Development Dependency Working Agreement
- [x] My code DOES NOT include the importing of new dependencies into the DPIA ecosystem
- [ ] My code DOES include the importing of new dependencies into the DPIA ecosystem
**If new dependencies are being introduced to the DPIA ecosystem:**
- [ ] The functionality of the dependency drastically reduces code complexity and makes my changes more easily maintainable and readible 
- [ ] The dependency being introduced does not contain multiple layers of nested dependencies introducing maintainability complexity to the DPIA ecosystem

## Frontend Development Changes
- [ ] N/A
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to project documentation or diagrams that reflect my changes
- [ ] New and existing unit tests pass locally with my changes
- [ ] My code follows Airbnb React Style Guidelines

## API Development Changes
- [x] N/A
- [ ] I have performed a self-review of my own code
- [ ] My code follows standards and practices outlined in the BC Government API Development Guidelines
- [ ] New and existing unit tests pass locally with my changes
- [ ] My changes includes Swagger documentation updates that reflect the changes I am introducing

## Definition of Done

Definition of Done

### CODE REVIEW

File: src/frontend/src/routes/router.tsx [hunk 1]
```diff
@@ -37,6 +37,7 @@ const Router = () => {
         <Route path="/pia-intake" element={<PIAIntakeFormPage />} />
         <Route path="/pia-result" element={<PIAIntakeResultsPage />} />
         <Route path="/pia-list" element={<PIAList />} />
+        <Route path="/pia/result/:id" element={<PIAList />} />
```
Comment (kushal-arora-fw) [inline src/frontend/src/routes/router.tsx:40]: `/pia/intake/:id/<title>`

title capped to 100 chars

Comment (kushal-arora-fw) [inline src/frontend/src/routes/router.tsx:40]: If the view page does not exist, no point keeping it to `PIAList`. Please add it when we have the view page ready.

**Review by kushal-arora-fw — CHANGES_REQUESTED** [review]
Added comments.

Plus the alignment of `View Details` button is off. Please rectify that too.

File: src/frontend/src/routes/router.tsx [hunk 2]
```diff
@@ -37,6 +37,10 @@ const Router = () => {
         <Route path="/pia-intake" element={<PIAIntakeFormPage />} />
         <Route path="/pia-result" element={<PIAIntakeResultsPage />} />
         <Route path="/pia-list" element={<PIAList />} />
+        <Route
+          path="/pia/result/:id/:title"
+          element={<div> not implemented yet </div>}
```
Comment (kushal-arora-fw) [inline src/frontend/src/routes/router.tsx:42]: is the `element` required field here? I was hoping to get rid of it altogether if not implemented.

File: src/frontend/src/routes/router.tsx [hunk 3]
```diff
@@ -37,6 +37,10 @@ const Router = () => {
         <Route path="/pia-intake" element={<PIAIntakeFormPage />} />
         <Route path="/pia-result" element={<PIAIntakeResultsPage />} />
         <Route path="/pia-list" element={<PIAList />} />
+        <Route
+          path="/pia/result/:id/:title"
```
Comment (kushal-arora-fw) [inline src/frontend/src/routes/router.tsx:41]: `/pia/intake/:id/<title>`

File: src/frontend/src/sass/_buttons.scss [hunk 4]
```diff
@@ -132,6 +132,7 @@ a:hover {
 .bcgovbtn__tertiary {
   --text-color: #{$primary-colour};
 
+  padding-top: 1.2em;
```
Comment (kushal-arora-fw) [inline src/frontend/src/sass/_buttons.scss:135]: adding padding-top to *all* secondary buttons isn't a way to go. Can you please come up with a neater solution.

Note: this impacts position of `List of PIAs` navigation buttons and other secondary btn usages.

**Review by kushal-arora-fw — CHANGES_REQUESTED** [review]
Left some comments

File: src/frontend/src/sass/_table.scss [hunk 5]
```diff
@@ -41,4 +41,8 @@
       }
     }
   }
+
+  .link {
+    padding-top: 1.2em;
+  }
```
Comment (kushal-arora-fw) [inline src/frontend/src/sass/_table.scss:47]: Now that you've removed the button, you can just replace line 38 with `& > .link` and get rid of these lines. And you should be good to go.

Comment (kushal-arora-fw) [reply src/frontend/src/sass/_table.scss:47]: but the problem is `link` does not follow the BEM naming convention. So, you could instead do
`& > a` and remove `link` class altogether from the <Link> element

Comment (binggaofw) [reply src/frontend/src/sass/_table.scss:47]: make the change, please check again

Comment (kushal-arora-fw) [reply src/frontend/src/sass/_table.scss]: I meant just drop in replacement for button to `a`. Let me suggest a commit. Let me know if that makes more sense to you.

File: src/frontend/src/components/public/PIAListTable/index.tsx [hunk 6]
```diff
@@ -54,13 +55,13 @@ const PIAListTable = ({ headings, pias }: IDataTable) => {
...
               <td>Submitted</td>
               <td>
-                <button
-                  className="bcgovbtn bcgovbtn__tertiary"
-                  onClick={() => handleDownload(pia.id)}
+                <Link
+                  target="_blank"
+                  rel="noopener noreferrer"
+                  to={`/pia/intake/${pia.id}/${pia.title}`}
                 >
```
Comment (kushal-arora-fw) [inline src/frontend/src/components/public/PIAListTable/index.tsx:62]:
```suggestion
                <Link
                  className="bcgovbtn bcgovbtn__tertiary"
                  target="_blank"
                  rel="noopener noreferrer"
                  to={`/pia/intake/${pia.id}/${pia.title}`}
                >
```

File: src/frontend/src/sass/_table.scss [hunk 7]
```diff
@@ -35,9 +35,9 @@
     & > tbody > tr > td:nth-child(5) {
       padding: 0;
 
-      & > button {
-        margin-inline: auto 0;
-        margin-top: 0.6em;
+      & > a {
+        padding-top: 1.2em;
+        display: block;
```
Comment (kushal-arora-fw) [inline src/frontend/src/sass/_table.scss:40]:
```suggestion
      & > a {
        margin-inline: auto 0;
        margin-top: 0.6em;
```

**Review by kushal-arora-fw — COMMENTED** [review]
Can you try if this works?

These will be minimal changes and please let me know if this aligns with what we need.

**Comment by binggaofw** [comment]
> Can you try if this works?
> 
> These will be minimal changes and please let me know if this aligns with what we need.

it does works
Screen Shot 2022-12-21 at 3 49 43 PM

**Review by kushal-arora-fw — APPROVED** [review]

**Pull request merged**